drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -78,6 +78,11 @@ func (c *portworxBasic) Reconcile(cluster *corev1.StorageCluster) error {
 	if err := c.createPortworxService(cluster, ownerRef); err != nil {
 		return NewError(ErrCritical, err)
 	}
+	if cluster.Spec.Kvdb != nil && cluster.Spec.Kvdb.Internal {
+		if err := c.createPortworxKVDBService(cluster, ownerRef); err != nil {
+			return NewError(ErrCritical, err)
```

Comment:
Actually, let's not return ErrCritical here, but instead just return err directly.
All above are necessary for portworx to function and so we want installation to fail if they fail, but the kvdb service is not critical.